platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/UserPropertiesDeletionScheduler.java

```diff
@@ -0,0 +1,150 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.insecure.defaults.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserPropertiesDeletionScheduler {
+
+  static final Path TEMP_FILE_TIMESTAMP_PATH =
+      Paths.get(new AbsolutePathResolver("data/tmp/timestamp.txt").getPath());
+  static final Path USERS_PROPERTIES_FILE_PATH =
+      Paths.get(new AbsolutePathResolver("etc/users.properties").getPath());
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(UserPropertiesDeletionScheduler.class);
+  private static boolean deletionScheduled;
+
+  public UserPropertiesDeletionScheduler() {
+    deletionScheduled = false;
+  }
+
+  public void scheduleDeletion(Path path) {
+    if (!deletionScheduled) {
+      String cron = getCron(path);
+      if (cron == null) {
+        return;
+      }
+      try {
+        CamelContext context = new DefaultCamelContext();
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .bean(UserPropertiesDeletionScheduler.class, "deleteFiles");
+              }
+            };
+
+        context.addRoutes(builder);
+        context.start();
+        deletionScheduled = true;
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create route", e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  String getCron(Path path) {
+    try {
+      if (!path.toFile().exists()) {
+        deleteFiles();
+        return null;
+      }
+
+      if (!TEMP_FILE_TIMESTAMP_PATH.toFile().exists()) {
+
+        File tempFile = new File(TEMP_FILE_TIMESTAMP_PATH.toString());
+        if (!tempFile.createNewFile()) {
+          return null;
+        }
+        Files.write(
+            TEMP_FILE_TIMESTAMP_PATH,
+            Long.toString(System.currentTimeMillis()).getBytes(StandardCharsets.UTF_8));
```

Comment:
✏️ It might be worth considering using an `ObjectOutputStream` to just write stream out the date object and then read it back in where you need it instead of doing all these conversions to long and Strings and dates yourself.